### ISSUE TITLE
Remove unnecessary non-null assertions in glean tests

### ISSUE THREAD

**Issue by jonalmeida** [mozilla-mobile/android-components#1105]
When you run unit tests locally, you see these warnings which can fix to clean up our build log.
```
./gradlew :service-glean:compileReleaseUnitTestKotlin
w: /Users/jalmeida/src/android-components/components/service/glean/src/test/java/mozilla/components/service/glean/storages/EventsStorageEngineTest.kt: (38, 47): Unnecessary non-null assertion (!!) on a non-null receiver of type List<EventsStorageEngine.RecordedEventData>
w: /Users/jalmeida/src/android-components/components/service/glean/src/test/java/mozilla/components/service/glean/storages/EventsStorageEngineTest.kt: (39, 60): Unnecessary non-null assertion (!!) on a non-null receiver of type List<EventsStorageEngine.RecordedEventData>
w: /Users/jalmeida/src/android-components/components/service/glean/src/test/java/mozilla/components/service/glean/storages/EventsStorageEngineTest.kt: (40, 55): Unnecessary non-null assertion (!!) on a non-null receiver of type List<EventsStorageEngine.RecordedEventData>
w: /Users/jalmeida/src/android-components/components/service/glean/src/test/java/mozilla/components/service/glean/storages/EventsStorageEngineTest.kt: (42, 29): Unnecessary non-null assertion (!!) on a non-null receiver of type List<EventsStorageEngine.RecordedEventData>
w: /Users/jalmeida/src/android-components/components/service/glean/src/test/java/mozilla/components/service/glean/storages/EventsStorageEngineTest.kt: (44, 29): Unnecessary non-null assertion (!!) on a non-null receiver of type List<EventsStorageEngine.RecordedEventData>
w: /Users/jalmeida/src/android-components/components/service/glean/src/test/java/mozilla/components/service/glean/storages/EventsStorageEngineTest.kt: (66, 47): Unnecessary non-null assertion (!!) on a non-null receiver of type List<EventsStorageEngine.RecordedEventData>
w: /Users/jalmeida/src/android-components/components/service/glean/src/test/java/mozilla/components/service/glean/storages/EventsStorageEngineTest.kt: (67, 62): Unnecessary non-null assertion (!!) on a non-null receiver of type List<EventsStorageEngine.RecordedEventData>
w: /Users/jalmeida/src/android-components/components/service/glean/src/test/java/mozilla/components/service/glean/storages/EventsStorageEngineTest.kt: (68, 55): Unnecessary non-null assertion (!!) on a non-null receiver of type List<EventsStorageEngine.RecordedEventData>
w: /Users/jalmeida/src/android-components/components/service/glean/src/test/java/mozilla/components/service/glean/storages/EventsStorageEngineTest.kt: (69, 48): Unnecessary non-null assertion (!!) on a non-null receiver of type List<EventsStorageEngine.RecordedEventData>
w: /Users/jalmeida/src/android-components/components/service/glean/src/test/java/mozilla/components/service/glean/storages/EventsStorageEngineTest.kt: (70, 81): Unnecessary non-null assertion (!!) on a non-null receiver of type List<EventsStorageEngine.RecordedEventData>
w: /Users/jalmeida/src/android-components/components/service/glean/src/test/java/mozilla/components/service/glean/storages/EventsStorageEngineTest.kt: (89, 43): Unnecessary non-null assertion (!!) on a non-null receiver of type List<EventsStorageEngine.RecordedEventData>
w: /Users/jalmeida/src/android-components/components/service/glean/src/test/java/mozilla/components/service/glean/storages/EventsStorageEngineTest.kt: (90, 49): Unnecessary non-null assertion (!!) on a non-null receiver of type List<EventsStorageEngine.RecordedEventData>
w: /Users/jalmeida/src/android-components/components/service/glean/src/test/java/mozilla/components/service/glean/storages/EventsStorageEngineTest.kt: (91, 51): Unnecessary non-null assertion (!!) on a non-null receiver of type List<EventsStorageEngine.RecordedEventData>
w: /Users/jalmeida/src/android-components/components/service/glean/src/test/java/mozilla/components/service/glean/storages/EventsStorageEngineTest.kt: (93, 25): Unnecessary non-null assertion (!!) on a non-null receiver of type List<EventsStorageEngine.RecordedEventData>
w: /Users/jalmeida/src/android-components/components/service/glean/src/test/java/mozilla/components/service/glean/storages/EventsStorageEngineTest.kt: (95, 25): Unnecessary non-null assertion (!!) on a non-null receiver of type List<EventsStorageEngine.RecordedEventData>
w: /Users/jalmeida/src/android-components/components/service/glean/src/test/java/mozilla/components/service/glean/storages/EventsStorageEngineTest.kt: (96, 54): Unnecessary non-null assertion (!!) on a non-null receiver of type List<EventsStorageEngine.RecordedEventData>
w: /Users/jalmeida/src/android-components/components/service/glean/src/test/java/mozilla/components/service/glean/storages/EventsStorageEngineTest.kt: (128, 40): Unnecessary non-null assertion (!!) on a non-null receiver of type List<EventsStorageEngine.RecordedEventData>
w: /Users/jalmeida/src/android-components/components/service/glean/src/test/java/mozilla/components/service/glean/storages/EventsStorageEngineTest.kt: (129, 47): Unnecessary non-null assertion (!!) on a non-null receiver of type List<EventsStorageEngine.RecordedEventData>
w: /Users/jalmeida/src/android-components/components/service/glean/src/test/java/mozilla/components/service/glean/storages/EventsStorageEngineTest.kt: (130, 48): Unnecessary non-null assertion (!!) on a non-null receiver of type List<EventsStorageEngine.RecordedEventData>
w: /Users/jalmeida/src/android-components/components/service/glean/src/test/java/mozilla/components/service/glean/storages/EventsStorageEngineTest.kt: (132, 22): Unnecessary non-null assertion (!!) on a non-null receiver of type List<EventsStorageEngine.RecordedEventData>
w: /Users/jalmeida/src/android-components/components/service/glean/src/test/java/mozilla/components/service/glean/storages/EventsStorageEngineTest.kt: (134, 22): Unnecessary non-null assertion (!!) on a non-null receiver of type List<EventsStorageEngine.RecordedEventData>
```

**Comment by karadkar** [comment]
Can I take this up ?

**Comment by jonalmeida** [comment]
@karadkar definitely! Feel free to provide a pull request for it with the commit message starting with: "Closes #1105: " to have it linked to this issue.

**Comment by karadkar** [comment]
@jonalmeida please check [1119](https://github.com/mozilla-mobile/android-components/pull/1119)
